### DISCUSSION TITLE
[Bug]: GSText maximum params

### DISCUSSION THREAD

**Discussion by npabisz** [OpenTTD/OpenTTD#12637] [Features]
### Version of OpenTTD

14.0

### Expected result

So currently there is some kind of old limit of 20 params in `GSText`. Which is very annoying especially for Town GUI, because it is single text. Eg. city builders shows requirements for towns to grow and right now it all needs to be done within one `GSText`. Solution could be either lift the limit of 20 params completely (or make it bigger like 100). Another solution would be rework of `GSTown.SetText` to work more like story pages where you can add new elements, multiple `GSText` instances with `GSStoryPage.NewElement`.

![Screenshot_1768](https://github.com/OpenTTD/OpenTTD/assets/13676368/863d86dd-533b-4f74-bbe6-f43ccbb880c4)

### Actual result

Error throwed if you use more than 20 params.

### Steps to reproduce

1. Set more than 20 params in GSText

**Comment by glx22** [comment]
The limitation comes from openttd string system itself.
